### PR TITLE
Shadow culling for v3.3.1

### PR DESCRIPTION
**Pull request by troublemaker52025**

### CODE REVIEW

**Comment by troublemaker52025** [comment]
https://github.com/cocos-creator/engine/pull/9151

**Comment by troublemaker52025** [comment]
https://github.com/cocos-creator/engine/pull/9152

File: cocos/math/Vec3.h [hunk 1]
```diff
@@ -51,22 +50,22 @@ class CC_DLL Vec3 {
     /**
      * The x-coordinate.
      */
-    float x;
+    float x{};
```
Comment (minggo) [inline cocos/math/Vec3.h:53]: float x{0.F} ？

File: cocos/math/Vec3.h [hunk 2]
```diff
@@ -188,6 +187,13 @@ class CC_DLL Vec3 {
      */
     void cross(const Vec3 &v);
 
+    /**
+     * Sets this vector to the cross product between itself and the specified vector.
+     *
+     * @param v1 The vector to compute the cross product with.
```
Comment (minggo) [inline cocos/math/Vec3.h:193]: 注释不太对

File: cocos/math/Vec3.h (hunk 1 above)
Comment (troublemaker52025) [reply cocos/math/Vec3.h:53]: member initializer for 'x' is redundant (modernize-use-default-member-init)

File: cocos/scene/Sphere.h [hunk 3]
```diff
@@ -52,11 +52,12 @@ class Sphere {
...
     int  spherePlane(const Plane &plane);
-    bool sphereFrustum(const Frustum &frustum);
+    bool sphereFrustum(const Frustum &frustum) const;
+    void mergeFrustum(const Frustum &frustum);
 
 private:
-    float _radius{0};
-    Vec3  _center;
+    float _radius{-1.0F};
+    Vec3  _center{};
```
Comment (minggo) [inline cocos/scene/Sphere.h:60]: Vec3  _center{} -> Vec3  _center ?
为什么需要加 {} 呢？

Comment (troublemaker52025) [reply cocos/scene/Sphere.h:60]: 没有必要写，顺手加的

**Comment by troublemaker52025** [comment]
需要金堃的审核通过才能合并

**Review by minggo — APPROVED** [review]

**Pull request merged**